### DISCUSSION TITLE
How do I print coraza's logs if I want to test with ftw

### DISCUSSION THREAD

**Discussion by Axkea** [corazawaf/coraza#853] [Q&A]
I'm currently building a plugin using coraza, and I need to use go-ftw to test whether the rule set works. Can you provide some examples of how I should filter the logs?

**Comment by dextermallo** [comment]
Hi @Axkea, what do you mean by filtering the logs? Do you mean you want to perform some actions after recording a log or do you what to know whether a rule is working against test cases from `go-ftw`?

**Reply by Axkea** [reply]
yes,I don't know how I can set up coraza's log and then give it to go-ftw to test

**Reply by Axkea** [reply]
wafConfig := coraza.NewWAFConfig()
			if len(config.auditLogPath) != 0 {
				wafConfig = wafConfig.WithErrorCallback(logError(config.auditLogPath))
			}

**Reply by Axkea** [reply]
This is how I set up the log now, but sometimes it doesn't have any output

**Reply by dextermallo** [reply]
I think the easiest way is to:
1. Start Coraza with CRS rules (e.g., https://github.com/corazawaf/coraza/tree/main/examples/http-server)

```go
// clone the rules folder from CRS and modify the Directives
waf, err := coraza.NewWAF(
		coraza.NewWAFConfig().
			WithErrorCallback(logError).
			WithDirectivesFromFile("./coreruleset/rules/*.conf").
			WithDirectivesFromFile("./coraza.conf"),
	)
```

2. Run the go-ftw, like `go-ftw run -d <Your_CRS_Folder/tests/regression/tests>`

In this way, you can have both `coraza log` and `go-ftw` in stdout, and you can also store it as a file if you wish

**Reply by Axkea** [reply]
I try to print this error log to a file but sometimes it doesn't work

**Reply by Axkea** [reply]
https://github.com/Axkea/envoy-go-waf/tree/main/ftw

**Reply by Axkea** [reply]
https://github.com/Axkea/envoy-go-waf/blob/main/plugin/parse.go#L117

**Comment by jcchavezs** [comment]
You can use this as inspiration
https://github.com/corazawaf/coraza-proxy-wasm/blob/main/ftw/docker-compose.yml

**Reply by Axkea** [reply]
I use this as a reference, but I need to know when the log is printed

**Reply by Axkea** [reply]
Because I need to output coraza's logs to a file, I can't set the log printer for coraza directly

**Reply by Axkea** [reply]
When I went to output coraza's log to a file, it worked under normal conditions, but it had no output when I ran the go-ftw test
